CONTRIBUTING.md

```diff
@@ -34,6 +35,46 @@ go get -u github.com/golang/dep/cmd/dep
 - [OpenPitrix](https://github.com/openpitrix/openpitrix): Application management platform on multi-cloud environment, it provides application template and application management for KubeSphere currently.
 - [SonarQube](https://github.com/SonarSource/sonarqube): Integrated in KubeSphere DevOps, it provides the capability to not only show health of an application but also to highlight issues newly introduced. 
 
+## Building KubeSphere on a local OS/shell environment
+
+### For Quick Taste Binary
+
+```bash
+mkdir ks-tmp
+cd ks-tmp
+echo 'module kubesphere' > go.mod
+echo 'replace (
+      	github.com/kiali/kiali => github.com/kubesphere/kiali v0.15.1-0.20190407071308-6b5b818211c3
+      	github.com/kubernetes-sigs/application => github.com/kubesphere/application v0.0.0-20190518133311-b9d9eb0b5cf7
+      )' >> go.mod
+      
+GO111MODULE=on go get kubesphere.io/kubesphere@cda4f4f05ae1d5c653e7158b8de0164f92341e5c
+GO111MODULE=on go build -o ks-apiserver kubesphere.io/kubesphere/cmd/ks-apiserver # build ks-apiserver
+GO111MODULE=on go build -o ks-apigateway kubesphere.io/kubesphere/cmd/ks-apigateway # build ks-apigateway
+GO111MODULE=on go build -o ks-controller-manager kubesphere.io/kubesphere/cmd/controller-manager # build ks-controller-manager
+GO111MODULE=on go build -o ks-iam kubesphere.io/kubesphere/cmd/ks-iam # build ks-iam
+```
+
+### For Building KubeSphere Images 
+
+KubeSphere components are often deployed as a container in a kubernetes cluster, you may need to build a Docker image locally.
+
+1. Clone repo to local
+
+```bash
+git clone https://github.com/kubesphere/kubesphere.git
+```
+
+2. Run Docker command to build image
+
+```bash
+docker build -f build/ks-apigateway/Dockerfile -t $REPO/ks-apigateway:$TAG .
```

Comment:
@runzexia  Does the "$Repo" means your personal github account? Could you pls add comments for $TAG? I am not sure for these values at the first time, it could be better that you can add comments in docs.